telemetry/definitions/commonDefinitions.json

```diff
@@ -517,6 +558,17 @@
             "description": "Called when updating a Lambda function's code outside the context of a SAM template",
             "metadata": [{ "type": "runtime", "required": false }, { "type": "result" }]
         },
+        {
+            "name": "lambda_deploy",
```

Comment:
What does this metric represent? Right now, we have two separate metrics for uploads:
* JB uses `lambda_editFunction` (JB can update both function configuration and code)
* VS Code will use `lambda_updateFunctionCode` 

Neither have a scratch-create Lambda. Is that what this represents?

Reply:
VS has configuration editing capabilities, which lambda_editFunction works for too.

JB and VS Code lean into SAM for their publishing operations, which is the sam_deploy metric. But they also allow a more one-off "update this code" operation. In VS, there is no SAM CLI related experience, rather its publishing operation is Lambda specific.

Reply:
OK, so if I understand this correctly, VS will emit both `editFunction` and `deploy`, with `deploy` occurring on both `Lambda.createFunction` and `Lambda.updateFunctionCode` calls (with `initialDeploy` true/false, respectively) and `editFunction` happening on `Lambda.updateFunctionConfiguration`?

If that's the case, we should standardize on those metrics sooner rather than later...could you cut an issue to make that change in JB? I can make the change to VS Code no sweat since Lambda upload is still in staging.

Reply:
I would suggest we don't need the granularity of "edit properties" vs "edit code", but otherwise if we are going to separate them, agree with @bryceitoc9 that the semantics should be consistent across IDEs.

Reply:
the nice thing about having separate metric for edit properties vs edit code - is that they are actually different workflows - if one is more popular than the other, and is missing as an option in other tools - it may drive us to change our functionality.

Reply:
Regardless, we have a potential redundancy: are `updateFunctionCode` and `deploy` with `initialDeploy: false` equivalent? If so, we should center on one (which would likely be `deploy` with `initialDeploy: false`) and we should do the following things:
* change VS Code to use `deploy`
* break apart Jetbrains' `editFunction` to also emit `deploy` if necessary (this seems like something we should do either way)